239 - Sample app

tmcconechy opened the pull request:
**Explain the details for making this change. What existing problem does the pull request solve?**

This PR adds a index.yml file to each component thats used to generate the contents of the kitchen sink page at https://www.figma.com/file/e1Q0W3ruWdTATmKxuwES1Z/demo-site-structure?node-id=0%3A1

For now just getting some basic functionality and the Yaml in. This wil be expanded out to this design https://www.figma.com/file/e1Q0W3ruWdTATmKxuwES1Z/demo-site-structure?node-id=0%3A1 but we dont have some of these components complete yet (header tabs ect)

Additional:

- update deps
- some cleanup on examples and renaming
- adds the ability to set a pallette color on ids-text (color="slate-50" for example)

**Related github/jira issue (required)**:
Fixes https://github.com/infor-design/enterprise-wc/issues/239 (for now)

**Steps necessary to review your pull request (required)**:
- pull this branch
- `npm prune && npm i && npm run start`
- go to http://localhost:4300/
- look for spelling / grammatical errors

EdwardCoyle commented on app/ids-toggle-button/index.yaml:
```diff
@@ -0,0 +1,15 @@
+---
+ component: Toogle Button
```
```suggestion
 component: Toggle Button
```

EdwardCoyle commented on app/ids-color/index.yaml:
```diff
@@ -0,0 +1,9 @@
+---
+ component: Color
+ link: ids-color
+ description: Color Swatchs
```
```suggestion
 description: Color Swatches
```

EdwardCoyle reviewed (commented):
Looks good so far!  I was wondering if we could use less columns for the clickable cards in the "Form Inputs", "Navigation and Interaction", and "Layouts" sections.  The items look a bit squished.  I think it might help to allow for more space inside each card.  In particular, this one seems like it could use the space:

![Screen Shot 2021-07-28 at 4 21 56 PM](https://user-images.githubusercontent.com/3249601/127390591-dcd16f19-51ed-446d-a514-518f1a277e09.png)

Other than that, just some corrections to the docs below.

rob2d commented on src/ids-text/ids-text.js:
```diff
@@ -61,6 +61,8 @@ class IdsText extends mix(IdsElement).with(
       attributes.FONT_SIZE,
       attributes.FONT_WEIGHT,
       attributes.LABEL,
+      attributes.MODE,
```
I think this one may have been an accident: there's a MODE included below (in 68)

rob2d commented on src/ids-text/ids-text.js:
```diff
@@ -61,6 +61,8 @@ class IdsText extends mix(IdsElement).with(
       attributes.FONT_WEIGHT,
       attributes.LABEL,
+      attributes.MODE,
+      attributes.OVERFLOW,
```
(similar to mode) this one is a few lines down again. I think these lines may have been a result of a merge since the ones there before were in the right alphabetical order.

rob2d reviewed (commented):
Just had one really tiny nit and one question (which might just stem from me not using the handlebars lib with webpack there), but everything looking overall really good @tmcconechy !

tmcconechy commented on webpack.config.js:
```diff
@@ -51,6 +57,11 @@ module.exports = {
   },
+  resolve: {
+    alias: {
+      handlebars: 'handlebars/dist/handlebars.js'
```
Removed this making the a mute point. This was something i tried and left in by accident and moved another direction so its not needed. Good catch

tmcconechy commented:
@EdwardCoyle @rob2d addressed the suggestions. For the "squished" issue I agree and realized i could use block grid instead. It seems to solve the layout issues and look a lot better check it out

EdwardCoyle reviewed (approved):
Looks a lot better!

rob2d reviewed (approved):
Agree, looks good!
